typing/printtyp.ml

```diff
@@ -1418,6 +1418,11 @@ let is_unit env ty =
   | Tconstr (p, _, _) -> Path.same p Predef.path_unit
   | _ -> false
 
+let is_ref_path p =
+  match p with
+  | Pdot (Pident id, "ref", _) -> Ident.same id ident_pervasives
```

Comment:
I believe that is what`non_shadowed_pervasive` is for, and it's more compatible with the `Stdlib` PR.